src/containers/Tasks/RequestAttestation/RequestAttestation.tsx

```diff
@@ -155,32 +156,51 @@ class RequestAttestation extends React.Component<
   private handleSubmit(): void {
     const {
       receiverAddresses,
-      legitimations,
+      terms,
       delegationId,
       onFinished,
+      quote,
     } = this.props
     const { savedClaimEntry } = this.state
 
     if (savedClaimEntry) {
-      attestationWorkflow
-        .requestAttestationForClaim(
-          savedClaimEntry.claim,
-          receiverAddresses,
-          (legitimations || []).map((legitimation: sdk.IAttestedClaim) =>
-            sdk.AttestedClaim.fromAttestedClaim(legitimation)
-          ),
-          delegationId
-        )
-        .then(() => {
-          if (onFinished) {
-            onFinished()
-          }
-        })
+      if (quote) {
+        attestationWorkflow
+          .requestAttestationForClaim(
+            savedClaimEntry.claim,
+            receiverAddresses,
+            (terms || []).map((legitimation: sdk.IAttestedClaim) =>
+              sdk.AttestedClaim.fromAttestedClaim(legitimation)
+            ),
+            delegationId,
+            quote
+          )
+          .then(() => {
+            if (onFinished) {
+              onFinished()
+            }
+          })
+      } else {
+        attestationWorkflow
+          .requestAttestationForClaim(
+            savedClaimEntry.claim,
+            receiverAddresses,
+            (terms || []).map((legitimation: sdk.IAttestedClaim) =>
+              sdk.AttestedClaim.fromAttestedClaim(legitimation)
+            ),
+            delegationId
+          )
+          .then(() => {
+            if (onFinished) {
+              onFinished()
+            }
+          })
+      }
```

Comment:
`requestAttestationForClaim()`'s parameter is `quoteAttesterSigned?: sdk.IQuoteAttesterSigned | undefined` and quote is also `quote?: sdk.IQuoteAttesterSigned | undefined`, so it seems like this could also just be 
```suggestion
      attestationWorkflow
        .requestAttestationForClaim(
          savedClaimEntry.claim,
          receiverAddresses,
          (terms || []).map((legitimation: sdk.IAttestedClaim) =>
            sdk.AttestedClaim.fromAttestedClaim(legitimation)
          ),
          delegationId,
          quote
        )
        .then(() => {
          if (onFinished) {
            onFinished()
          }
        })
```
Or if that's relying too much on static type checking, this should be equivalent:
```suggestion
      attestationWorkflow
        .requestAttestationForClaim(
          savedClaimEntry.claim,
          receiverAddresses,
          (terms || []).map((legitimation: sdk.IAttestedClaim) =>
            sdk.AttestedClaim.fromAttestedClaim(legitimation)
          ),
          delegationId,
          quote || undefined
        )
        .then(() => {
          if (onFinished) {
            onFinished()
          }
        })
```

Reply:
I do prefer the second approach, but I don't understand what the difference technically, only the visual and understanding of the code. 

I understand the reasoning, just not what the two differences. What are the benefits of having the undefined in one place over the other? If you are willing to share the reasoning 👍  

It looks much better and cleaner code.

Reply:
The differences between the two: The first one relies on the possible types of `quote` being as per type definition, namely `sdk.IQuoteAttesterSigned | undefined`, while the second additionally handles unexpected falsy types by replacing them with `undefined`. Which one did you pick?

Reply:
Inside the `requestAttestationForClaim:{...., quote || undefined...}`